### ISSUE TITLE
beacons.service: indices must be integers, not str (Windows)

### ISSUE THREAD

**Issue by jlayze** [saltstack/salt#41737]
### Description of Issue/Question
I'm trying to setup a beacon to fire an event when the Print Spooler status changes, and then a reactor to start the Print Spooler service if it's not running.  The end goal is to do this for a different service, but I'm using the Print Spooler for testing.  However, I am getting an error that keeps looping every 1 second:

```
[CRITICAL] The beacon errored:
Traceback (most recent call last):
  File "C:\salt\bin\lib\site-packages\salt\minion.py", line 2221, in handle_beacons
    beacons = self.process_beacons(self.functions)
  File "C:\salt\bin\lib\site-packages\salt\minion.py", line 420, in process_beacons
    return self.beacons.process(b_conf, self.opts['grains'])  # pylint: disable=no-member
  File "C:\salt\bin\lib\site-packages\salt\beacons\__init__.py", line 100, in process
    raw = self.beacons[fun_str](b_config[mod])
  File "C:\salt\bin\lib\site-packages\salt\beacons\service.py", line 91, in beacon
    config[service]['oncleanshutdown'] = False
TypeError: list indices must be integers, not str
```

Also, in beacons.service from lines 87 to 91, I think all references to 'oncleanshutdown' should be replaced with 'uncleanshutdown'.

### Setup
/srv/salt/Top.sls
```
service:
  beacon.present:
    - "Spooler":
      - onchangeonly: True
```
/conf/master.d/reactor.conf
```
reactor:
  - salt/beacon/*/service/"Spooler"
    - /srv/reactor/SpoolerReactor.sls
```
/srv/reactor/SpoolerReactor.sls
```
SpoolerRestart:
  local.state.apply:
    - tgt: {{ data['data']['id'] }}
    - arg:
      - SpoolerStart
```
/srv/salt/SpoolerStart.sls
```
"Spooler":
  service.running:
    - enable: True
```

### Steps to Reproduce Issue
Just running `salt "minion-id" state.apply` will start the looping error on the minion.

### Versions Report
Both Master and Minion running Windows Server 2012 R2.

```
Salt Version:
           Salt: 2016.11.5
 
Dependency Versions:
           cffi: 1.10.0
       cherrypy: unknown
       dateutil: 2.5.3
      docker-py: Not Installed
          gitdb: 0.6.4
      gitpython: 2.0.8
          ioflo: 1.5.5
         Jinja2: 2.9.4
        libgit2: Not Installed
        libnacl: 1.4.5
       M2Crypto: Not Installed
           Mako: 1.0.4
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: Not Installed
      pycparser: 2.17
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.12 (v2.7.12:d33e0cf91556, Jun 27 2016, 15:24:40) [MSC v.1500 64 bit (AMD64)]
   python-gnupg: 0.3.8
         PyYAML: 3.11
          PyZMQ: 16.0.1
           RAET: Not Installed
          smmap: 0.9.0
        timelib: 0.2.4
        Tornado: 4.4.1
            ZMQ: 4.1.6
 
System Versions:
           dist:   
        machine: AMD64
        release: 2012ServerR2
         system: Windows
        version: 2012ServerR2 6.3.9600  Multiprocessor Free
```

**Comment by garethgreenaway** [comment]
@jlayze Thanks for the report.  The format for beacons changed and it looks like the state module to manage them was not updated to take that into account.  We'll get this fixed.

**Comment by garethgreenaway** [comment]
@jlayze Update your state file to look like this:
```
service:
  beacon.present:
    - "Spooler":
        onchangeonly: True
````

**Comment by jlayze** [comment]
@garethgreenaway I'm no longer getting an error after making this change, but now I'm not sure if my reactor.conf or reactor.sls files are formatted correctly.

When I run `salt "wincse-labts1" beacons.list`:
```
wincse-labts1:
    beacons:
      service:
        Spooler:
          onchangeonly: true
```

When I start or stop the Print Spooler, this is what I see on the Master:
```
[DEBUG   ] Sending event: tag = salt/beacon/wincse-labts1/service/; data = {'service_name': 'Spooler', 
 _stamp': '2017-06-15T13:36:04.536000', 'Spooler': {'running': False}, 'id': 'wincse-labts1'}
[DEBUG   ] Gathering reactors for tag salt/beacon/wincse-labts1/service/
```
/conf/master.d/reactor.conf:
```
reactor:
  - salt/beacon/*/service/Spooler:
    - /srv/reactor/SpoolerReactor.sls
```
/srv/reactor/SpoolerReactor.sls:
```
SpoolerRestart:
  local.state.apply:
    - tgt: {{ data['id'] }}
    - arg:
      - SpoolerStart
```
#
I also tried an alternative setup, which has the same results:
/conf/master.d/reactor.conf:
```
reactor:
  - salt/beacon/*/service:
    - /srv/reactor/ServiceReactor.sls
```
/srv/reactor/ServiceReactor.sls:
```
{% if data['service_name'] == 'Spooler' %}
SpoolerRestart:
  local.state.apply:
    - tgt: {{ data['id'] }}
    - arg:
      - SpoolerStart
{% endif %}
```

I tried using {{ data['data']['id'] }} as well based on documentation [here](https://docs.saltstack.com/en/latest/topics/beacons/index.html#view-events-on-the-master) and [here](https://docs.saltstack.com/en/latest/topics/reactor/index.html#beacons-and-reactors) but saw no difference.  Not sure what I'm doing wrong, I'm getting the event just need the master to react.

**Comment by jlayze** [comment]
I figured it out... needed a trailing `/` for `'salt/beacon/*/service/'` because that's what the event returns as the tag.  And then for the path to the reactor I needed to start with `/salt/srv/` **not** `/srv/` (this might only relate to Windows masters)

/conf/master.d/reactor.conf:
```
reactor:
  - 'salt/beacon/*/service/':
    - /salt/srv/reactor/ServiceReactor.sls
```
/srv/reactor/ServiceReactor.sls:
```
{% if data['service_name'] == 'Spooler' %}
SpoolerRestart:
  local.state.apply:
    - tgt: {{ data['id'] }}
    - arg:
      - SpoolerStart
{% endif %}
```
Now stopping the Print Spooler results in it starting right back up again.
#
It would appear the [documentation](https://docs.saltstack.com/en/latest/topics/reactor/index.html#beacons-and-reactors) is wrong about `{{ data['data']['id'] }}`

**Comment by garethgreenaway** [comment]
Fixed in #42317
